src/colmap/scene/image.h

```diff
@@ -124,8 +140,13 @@ class Image {
   // where `point3D_id != kInvalidPoint3DId`.
   point2D_t num_points3D_;
 
+  // [Optional] The corresponding frame (rig) of the image
+  frame_t frame_id_;
+  std::shared_ptr<class Frame> frame_ = nullptr;
+
   // The pose of the image, defined as the transformation from world to camera.
-  Rigid3d cam_from_world_;
+  // Only useful when the corresponding frame (rig) does not exist.
+  mutable Rigid3d cam_from_world_;
```

Comment:
Let's avoid mutable here?

Reply:
My initial thought is to do this: https://github.com/colmap/colmap/pull/2698/files#diff-665386c3e6e1ebf0f2b3f383a2925c7a970b2e256f2f889f27fb090065158fbdR210, but indeed this feels unnecessary. Then we may want to explicitly mention in the comments somewhere that the member ``cam_from_world_`` is not used when ``frame_`` is available. 

Note that with the current rig support we cannot do anything as easy as ``image.CamfromWorld() = cam_from_world_prior`` anymore when ``frame_`` is available, which actually makes sense but can be concerning.

Reply:
Thanks, I would avoid such magic and make it more explicit. I only very rarely use "mutable" and in 99% of those rare cases, I use it for std::mutex members.

Reply:
Its unclear how to return a const reference here without mutuble: https://github.com/colmap/colmap/pull/2698/files#diff-665386c3e6e1ebf0f2b3f383a2925c7a970b2e256f2f889f27fb090065158fbdR211. Any ideas?